File: app/src/main/java/com/laundromat/pos/data/model/Order.kt

package com.laundromat.pos.data.model
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "orders",
    foreignKeys = [
        ForeignKey(
            entity = Machine::class,
            parentColumns = ["id"],
            childColumns = ["machineId"],
            onDelete = ForeignKey.RESTRICT
        ),
        ForeignKey(
            entity = Customer::class,
            parentColumns = ["id"],
            childColumns = ["customerId"],
            onDelete = ForeignKey.SET_NULL
        ),
        ForeignKey(
            entity = PosUser::class,
            parentColumns = ["id"],
            childColumns = ["createdByUserId"],
            onDelete = ForeignKey.SET_NULL
        )
    ],
    indices = [
        Index("machineId"),
        Index("customerId"),
        Index("createdByUserId")
    ]
)
data class Order(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val machineId: Int,
    val customerId: Int? = null,
    val createdByUserId: Int? = null,
    val serviceType: String = "WASH",
    val amount: Double,
    val status: String = OrderStatus.OPEN,
    val paymentMethod: String,
    val paymentStatus: String = PaymentStatus.PENDING,
    val paymentReference: String? = null,
    val notes: String = "",
    val timestamp: Long = System.currentTimeMillis()
)

object OrderStatus {
    const val OPEN = "OPEN"
    const val IN_PROGRESS = "IN_PROGRESS"
    const val COMPLETED = "COMPLETED"
    const val CANCELLED = "CANCELLED"
}

object PaymentStatus {
    const val PENDING = "PENDING"
    const val PAID = "PAID"
    const val FAILED = "FAILED"
    const val REFUNDED = "REFUNDED"
}

File: flutter_app/lib/src/models/order.dart
class OrderStatus {
  static const booked = 'BOOKED';
  static const inProgress = 'IN_PROGRESS';
  static const completed = 'COMPLETED';
}

class PaymentStatus {
  static const pending = 'PENDING';
  static const paid = 'PAID';
  static const refunded = 'REFUNDED';
}

class Order {
  const Order({
    required this.id,
    required this.machineId,
    required this.customerId,
    required this.createdByUserId,
    required this.serviceType,
    required this.amount,
    required this.status,
    required this.paymentMethod,
    required this.paymentStatus,
    required this.paymentReference,
    required this.timestamp,
    this.loadSizeKg,
    this.washOption,
    this.dryerMachineId,
  });

  final int id;
  final int machineId;
  final int customerId;
  final int? createdByUserId;
  final String serviceType;
  final double amount;
  final String status;
  final String paymentMethod;
  final String paymentStatus;
  final String paymentReference;
  final DateTime timestamp;
  final int? loadSizeKg;
  final String? washOption;
  final int? dryerMachineId;

  Order copyWith({
    int? id,
    int? machineId,
    int? customerId,
    Object? createdByUserId = _sentinel,
    String? serviceType,
    double? amount,
    String? status,
    String? paymentMethod,
    String? paymentStatus,
    String? paymentReference,
    DateTime? timestamp,
    Object? loadSizeKg = _sentinel,
    Object? washOption = _sentinel,
    Object? dryerMachineId = _sentinel,
  }) {
    return Order(
      id: id ?? this.id,
      machineId: machineId ?? this.machineId,
      customerId: customerId ?? this.customerId,
      createdByUserId: identical(createdByUserId, _sentinel)
          ? this.createdByUserId
          : createdByUserId as int?,
      serviceType: serviceType ?? this.serviceType,
      amount: amount ?? this.amount,
      status: status ?? this.status,
      paymentMethod: paymentMethod ?? this.paymentMethod,
      paymentStatus: paymentStatus ?? this.paymentStatus,
      paymentReference: paymentReference ?? this.paymentReference,
      timestamp: timestamp ?? this.timestamp,
      loadSizeKg: identical(loadSizeKg, _sentinel)
          ? this.loadSizeKg
          : loadSizeKg as int?,
      washOption: identical(washOption, _sentinel)
          ? this.washOption
          : washOption as String?,
      dryerMachineId: identical(dryerMachineId, _sentinel)
          ? this.dryerMachineId
          : dryerMachineId as int?,
    );
  }
}

const _sentinel = Object();
